DigitalLearningSolutions.Data.Tests/Services/TutorialContentServiceTests.cs

```diff
@@ -582,6 +666,402 @@ public void Get_tutorial_information_should_have_default_course_settings_when_js
+
+            // Then
+            tutorial.Should().NotBeNull();
+            tutorial!.OtherItemsInSectionExist.Should().BeTrue();
+        }
+
+        [Test]
+        public void Get_tutorial_information_has_no_other_items_in_section()
+        {
+            // Given
+            const int candidateId = 272596;
+            const int customisationId = 23048;
+            const int sectionId = 2027;
+            const int tutorialId = 9526;
+
+            // When
+            var tutorial = tutorialContentService.GetTutorialInformation(candidateId, customisationId, sectionId, tutorialId);
+
+            // Then
+            tutorial.Should().NotBeNull();
+            tutorial!.OtherItemsInSectionExist.Should().BeFalse();
+        }
+
+        [Test]
+        public void Get_tutorial_information_has_no_other_items_in_section_if_other_tutorials_are_archived()
+        {
+            using (new TransactionScope())
+            {
+                // Given
+                const int customisationId = 14895;
+                const int candidateId = 22045;
+                const int sectionId = 333;
+                const int tutorialId = 1339;
+
+                // The tutorials in this section, which does not have a post learning assessment or consolidation path
+                tutorialContentTestHelper.ArchiveTutorial(1340);
+                tutorialContentTestHelper.ArchiveTutorial(1341);
+
+                // When
+                var tutorial = tutorialContentService.GetTutorialInformation(candidateId, customisationId, sectionId, tutorialId);
+
+                // Then
+                tutorial.Should().NotBeNull();
+                tutorial!.OtherItemsInSectionExist.Should().BeFalse();
+            }
+        }
+
+        [Test]
+        public void Get_tutorial_information_has_other_items_in_section_if_another_tutorial_has_diagnostic()
+        {
+            using (new TransactionScope())
+            {
+                // Given
+                const int customisationId = 5468;
+                const int candidateId = 94705;
+                const int sectionId = 102;
+                const int tutorialId = 316;
+
+                // Remove diagnostic from this tutorial
+                sectionContentTestHelper.UpdateDiagnosticStatus(tutorialId, customisationId, 0);
+
+                // Make other tutorials in this section inaccessible, but they still have a diagnostic status 1
+                tutorialContentTestHelper.UpdateTutorialStatus(317, customisationId, 0);
+                tutorialContentTestHelper.UpdateTutorialStatus(318, customisationId, 0);
+
+                // When
+                var tutorial = tutorialContentService.GetTutorialInformation(candidateId, customisationId, sectionId, tutorialId);
+
+                // Then
+                tutorial.Should().NotBeNull();
+                tutorial!.OtherItemsInSectionExist.Should().BeFalse();

```

Comment:
Shouldn't this be true? 😕

Reply:
Yep, and there was a bug there as well! Thanks for that, fixed in 3f3f8da 🙂